Tests/LeafErrorMiddlewareTests/CustomMappingCustomGeneratorTests.swift

```diff
@@ -0,0 +1,148 @@
+                .notFound
+            }
+
+            router.get("403") { _ -> Response in
+                throw Abort(.forbidden)
+            }
+
+            router.get("serverError") { _ -> Response in
+                throw Abort(.internalServerError)
+            }
+
+            router.get("unknownError") { _ -> Response in
+                throw TestError()
+            }
+
+            router.get("unauthorized") { _ -> Response in
+                throw Abort(.unauthorized)
+            }
+
+            router.get("303") { _ -> Response in
+                throw Abort.redirect(to: "ok")
+            }
+
+            router.get("404withReason") { _ -> HTTPStatus in
+                throw Abort(.notFound, reason: "Could not find it")
+            }
+
+            router.get("500withReason") { _ -> HTTPStatus in
+                throw Abort(.badGateway, reason: "I messed up")
+            }
+        }
+
+        try routes(app)
+        let mappings: [HTTPStatus: String] = [
+            .notFound: "404",
+            .unauthorized: "401",
+            .forbidden: "403",
+            // Verify that non-error mappings are ignored
+            .seeOther: "303"
+        ]
+        let leafMiddleware = LeafErrorMiddleware(errorMappings: mappings) { status, error, req async throws -> AContext in
+            AContext(trigger: true)
+        }
+        app.middleware.use(leafMiddleware)
+    }
+
+    override func tearDownWithError() throws {
+        app.shutdown()
+        try eventLoopGroup.syncShutdownGracefully()
+    }
+
+    // MARK: - Tests
+
+    func testThatValidEndpointWorks() throws {
+        let response = try app.getResponse(to: "/ok")
+        XCTAssertEqual(response.status, .ok)
+    }
+
+    func testThatRedirectIsNotCaught() throws {
+        let response = try app.getResponse(to: "/303")
+        XCTAssertEqual(response.status, .seeOther)
+        XCTAssertEqual(response.headers[.location].first, "ok")
+        XCTAssertNotEqual(viewRenderer.leafPath, "303")
+    }
+
+    func testNonAbort404IsCaughtCorrectly() throws {
+        let response = try app.getResponse(to: "/404")
+        XCTAssertEqual(response.status, .notFound)
+        XCTAssertEqual(viewRenderer.leafPath, "404")
+    }
+
+    func testThat403IsCaughtCorrectly() throws {
+        let response = try app.getResponse(to: "/403")
+        XCTAssertEqual(response.status, .forbidden)
+        XCTAssertEqual(viewRenderer.leafPath, "403")
+        let context = try XCTUnwrap(viewRenderer.capturedContext as? AContext)
+        XCTAssertTrue(context.trigger)
+    }
+
+    func testContextGeneratedOn500Page() throws {
+        let response = try app.getResponse(to: "/unauthorized")
+        XCTAssertEqual(response.status, .unauthorized)
+        XCTAssertEqual(viewRenderer.leafPath, "401")
+        let context = try XCTUnwrap(viewRenderer.capturedContext as? AContext)
+        XCTAssertTrue(context.trigger)
+    }
+
+    func testGetAResponseWhenGenerator() throws {
```

Comment:
Nit - it's not clear what we're testing here. It looks like we're missing something after "when generator <SOMETHING>"